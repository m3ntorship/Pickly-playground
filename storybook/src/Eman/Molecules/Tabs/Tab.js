import React, { useState } from "react";
import PropTypes from "prop-types";
import { Radio } from "./../../Atoms/Radios/Radio";
import Image from "./../../assets/icons/Tab/Image";
import Text from "./../../assets/icons/Tab/Text";
import File from "./../../assets/icons/Tab/File";

export const TAB = {
  ICON: {
    IMAGE: "Image",
    TEXT: "Text",
    MORE: "More",
    NONE: "None",
  },
  STYLES: {
    ACTIVE: "Active",
    INACTIVE: "Inactive",
    DISABLED: "Disabled",
  },
};

export const Tab = ({ content, icon, styles }) => {
  const [clicked, setClicked] = useState(false);

  const style =
    styles === TAB.STYLES.ACTIVE
      ? "border border-grey px-4 h-8 py-1.5 rounded-full font-semibold"
      : styles === TAB.STYLES.INACTIVE
      ? "bg-grey-bg px-4 h-8 py-1.5 rounded-full"
      : "bg-grey-shd6 px-4 h-8 py-1.5 rounded-full opacity-60";

  const handleClick = () => {
    setTimeout(() => setClicked(!clicked));
  };

  return (
    <>
      <label
        className={`inline-block ${!clicked && style} ${
          clicked &&
          "border border-grey px-4 h-8 py-1.5 rounded-full font-semibold"
        } ml-2 hover:bg-grey-shd6 hover:px-4 hover:h-8 hover:py-1.5 hover:rounded-full hover:border-grey-shd6 focus:outline-none cursor-pointer`}
        onClick={handleClick}
      >
        <div className="flex items-center">
          <Radio size="Small" />
          {icon === TAB.ICON.IMAGE && <Image className="ml-2" />}
          {icon === TAB.ICON.TEXT && <Text className="ml-2" />}
          {icon === TAB.ICON.MORE && <File className="ml-2" />}
          {icon === TAB.ICON.NONE && ""}
          <p className="text-sm text-dark-grey my-0 ml-2">{content}</p>
        </div>
      </label>
    </>
  );
};

Tab.propTypes = {
  icon: PropTypes.oneOf(["Image", "Text", "More", "None"]),
  content: PropTypes.string,
  styles: PropTypes.oneOf(["Active", "Hover", "Inactive", "Disabled"]),
};
