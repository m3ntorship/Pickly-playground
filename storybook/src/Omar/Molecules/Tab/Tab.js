import { React, useState } from "react";
import PropTypes from "prop-types";
export const SIZE = {
  BUTTONSIZE: {
    DEFAULT: "default",
    SMALL: "small",
  },
  RADIOSIZE: {
    DEFAULT: "default",
    SMALL: "small",
  },
};

export const ICONS = {
  IMAGE: "image",
  TEXT: "text",
  DOCUMENT: "document",
};
export const Tab = (props) => {
  const { disabled, ButtonSize, LabelText, Icons } = props;
  const [isChecked, setIsChecked] = useState(false);
  const handleCheck = () => {
    setIsChecked(!isChecked);
  };
  return (
    <button
      className={`focus:outline-none mr-5 ${
        ButtonSize === SIZE.BUTTONSIZE.SMALL ? "w-28 h-2" : "w-32 h-4"
      }`}
    >
      <label
        htmlFor="radio-label1"
        className={`relative cursor-pointer flex flex-row justify-around items-center border border-grey rounded-full py-1.5 pr-3.5 pl-3 hover:bg-grey-shd6 ${
          disabled && "bg-grey-shd6 border-none opacity-60"
        }`}
      >
        <>
          {isChecked && (
            <label
              htmlFor="radio-label1"
              className={` relative rounded-lg bg-primary flex justify-center items-center hover:bg-primary-shd8 focus:ring-2 focus:ring-offset-primary-shd6 focus:outline-none ${
                ButtonSize === SIZE.BUTTONSIZE.SMALL ? "w-4 h-4" : "w-5 h-5"
              } ${disabled && "bg-primary-shd7"}`}
              onClick={handleCheck}
            >
              <input
                type="radio"
                className={`opacity-0 absolute cursor-pointer w-full h-full ${
                  disabled && "cursor-not-allowed"
                }`}
                id="radio-label1"
                name="radio"
                checked={disabled}
              />
              <div
                className={`${
                  ButtonSize === SIZE.BUTTONSIZE.SMALL
                    ? "w-1.5 h-1.5"
                    : "w-2 h-2"
                } rounded-lg bg-white`}
              ></div>
            </label>
          )}
          {!isChecked && (
            <label
              htmlFor="radio-label1"
              className={` relative border border-grey-shd4 rounded-lg flex justify-center items-center hover:border-grey-shd1 focus:ring-2 focus:border-offset-grey-shd2 focus:outline-none w-4 h-4 ${
                ButtonSize === SIZE.BUTTONSIZE.SMALL ? "w-4 h-4" : "w-5 h-5"
              } ${disabled && "border-2 border-grey-shd6"}`}
              onClick={handleCheck}
            >
              <input
                type="radio"
                className={`opacity-0 absolute cursor-pointer w-full h-full ${
                  disabled && "cursor-not-allowed"
                }`}
                id="radio-label1"
                name="radio"
                checked={disabled}
              />
            </label>
          )}
        </>
        <>
          {Icons === ICONS.IMAGE ? (
            <svg
              className="mt-0.5"
              width="16"
              height="16"
              viewBox="0 0 16 16"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.99927 7.3335C5.55155 7.3335 5.99927 6.88578 5.99927 6.3335C5.99927 5.78121 5.55155 5.3335 4.99927 5.3335C4.44698 5.3335 3.99927 5.78121 3.99927 6.3335C3.99927 6.88578 4.44698 7.3335 4.99927 7.3335Z"
                fill="#5B6366"
              />
              <path
                d="M6.99927 9.33317L5.99927 7.99984L3.99927 10.6665H5.99927H7.99927H11.9993L8.99927 6.6665L6.99927 9.33317Z"
                fill="#5B6366"
              />
              <path
                d="M13.3328 2.6665H2.6661C1.93076 2.6665 1.33276 3.2645 1.33276 3.99984V11.9998C1.33276 12.7352 1.93076 13.3332 2.6661 13.3332H13.3328C14.0681 13.3332 14.6661 12.7352 14.6661 11.9998V3.99984C14.6661 3.2645 14.0681 2.6665 13.3328 2.6665ZM2.6661 11.9998V3.99984H13.3328L13.3341 11.9998H2.6661Z"
                fill="#5B6366"
              />
            </svg>
          ) : Icons === ICONS.TEXT ? (
            <svg
              width="16"
              height="16"
              viewBox="0 0 16 16"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.6665 7.33333H9.33317V8.66666H4.6665V7.33333ZM4.6665 4.66666H11.9798V6H4.6665V4.66666ZM4.6665 10H13.3332V11.3333H4.6665V10ZM2.6665 2.66666H3.99984V13.3333H2.6665V2.66666Z"
                fill="#5B6366"
              />
            </svg>
          ) : (
            <svg
              width="16"
              height="16"
              viewBox="0 0 16 16"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.2685 5.724C13.2358 5.65334 13.1952 5.586 13.1378 5.52867L9.13784 1.52867C9.0805 1.47134 9.01317 1.43067 8.9425 1.398C8.9225 1.38867 8.90117 1.38334 8.87984 1.376C8.82384 1.35734 8.7665 1.34534 8.70717 1.342C8.69317 1.34067 8.6805 1.33334 8.6665 1.33334H3.99984C3.2645 1.33334 2.6665 1.93134 2.6665 2.66667V13.3333C2.6665 14.0687 3.2645 14.6667 3.99984 14.6667H11.9998C12.7352 14.6667 13.3332 14.0687 13.3332 13.3333V6C13.3332 5.986 13.3258 5.97334 13.3245 5.95867C13.3212 5.89934 13.3098 5.842 13.2905 5.786C13.2838 5.76467 13.2778 5.744 13.2685 5.724ZM11.0572 5.33334H9.33317V3.60934L11.0572 5.33334ZM3.99984 13.3333V2.66667H7.99984V6C7.99984 6.36867 8.29784 6.66667 8.6665 6.66667H11.9998L12.0012 13.3333H3.99984Z"
                fill="#5B6366"
              />
              <path
                d="M5.3335 8H10.6668V9.33334H5.3335V8ZM5.3335 10.6667H10.6668V12H5.3335V10.6667ZM5.3335 5.33334H6.66683V6.66667H5.3335V5.33334Z"
                fill="#5B6366"
              />
            </svg>
          )}
        </>
        <span
          className={`text-dark-grey ${
            ButtonSize === SIZE.BUTTONSIZE.SMALL ? "text-sm" : "text-base"
          }`}
        >
          {LabelText}
        </span>
      </label>
    </button>
  );
};

Tab.propTypes = {
  disabled: PropTypes.bool,
  LabelText: PropTypes.string,
  ButtonSize: PropTypes.oneOf(["default", "small"]),
  Icons: PropTypes.oneOf(["image", "text", "document"]),
};
