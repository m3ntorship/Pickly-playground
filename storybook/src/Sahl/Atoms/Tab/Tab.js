import React from "react";
import PropTypes from "prop-types";
import Radio from "../Radio/Radio";

export default function Tab(props) {
  const {
    disabled = false,
    label,
    type,
    onClick,
    pollType,
    className,
    selected,
    value,
  } = props;

  // Style Classes
  let buttonClasses = [
    "h-19",
    "flex",
    "items-center",
    "rounded-full",
    "py-1.5",
    "px-7",
    "focus:outline-none",
    "hover:bg-grey-shd6",
  ];
  disabled ? buttonClasses.push("disabled:opacity-30") : buttonClasses.push("");

  // Handlers

  if (selected) {
    console.log(selected, "[Tab.js]");
    buttonClasses = [
      ...buttonClasses,
      "bg-white",
      "border",
      "border-solid",
      "border-grey-dark",
    ];
  } else {
    buttonClasses = [...buttonClasses, "bg-grey-bg"];
  }

  // Markup
  const pollElements = (
    <>
      <Radio
        value={value}
        name={`${type} tab`}
        checked={selected}
        onChange={onClick}
        className="mr-4"
      />
      {/* ////// Image /////// */}
      {pollType === "image" && (
        <svg
          className="w-4 h-4 mr-2"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.49902 11C8.32745 11 8.99902 10.3284 8.99902 9.5C8.99902 8.67157 8.32745 8 7.49902 8C6.6706 8 5.99902 8.67157 5.99902 9.5C5.99902 10.3284 6.6706 11 7.49902 11Z"
            fill="#5B6366"
          />
          <path
            d="M10.499 14L8.99902 12L5.99902 16H8.99902H11.999H17.999L13.499 10L10.499 14Z"
            fill="#5B6366"
          />
          <path
            d="M19.999 4H3.99902C2.89602 4 1.99902 4.897 1.99902 6V18C1.99902 19.103 2.89602 20 3.99902 20H19.999C21.102 20 21.999 19.103 21.999 18V6C21.999 4.897 21.102 4 19.999 4ZM3.99902 18V6H19.999L20.001 18H3.99902Z"
            fill="#5B6366"
          />
        </svg>
      )}
      {pollType === "text" && (
        <svg
          className="w-4 h-4 mr-2"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 11H14V13H7V11ZM7 7H17.97V9H7V7ZM7 15H20V17H7V15ZM4 4H6V20H4V4Z"
            fill="#5B6366"
          />
        </svg>
      )}
      {pollType === "survey" && (
        <svg
          className="w-4 h-4 mr-2"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.903 8.586C19.854 8.48 19.793 8.379 19.707 8.293L13.707 2.293C13.621 2.207 13.52 2.146 13.414 2.097C13.384 2.083 13.352 2.075 13.32 2.064C13.236 2.036 13.15 2.018 13.061 2.013C13.04 2.011 13.021 2 13 2H6C4.897 2 4 2.897 4 4V20C4 21.103 4.897 22 6 22H18C19.103 22 20 21.103 20 20V9C20 8.979 19.989 8.96 19.987 8.938C19.982 8.849 19.965 8.763 19.936 8.679C19.926 8.647 19.917 8.616 19.903 8.586ZM16.586 8H14V5.414L16.586 8ZM6 20V4H12V9C12 9.553 12.447 10 13 10H18L18.002 20H6Z"
            fill="#5B6366"
          />
          <path
            d="M8 12H16V14H8V12ZM8 16H16V18H8V16ZM8 8H10V10H8V8Z"
            fill="#5B6366"
          />
        </svg>
      )}
    </>
  );

  return (
    <button
      value={value}
      onClick={onClick}
      disabled={disabled}
      className={`${buttonClasses.join(" ")} ${className}`}
    >
      {type === "poll" && pollElements}
      <span className="text-body-small text-dark-grey capitalize">{label}</span>
    </button>
  );
}

Tab.propTypes = {
  type: PropTypes.oneOf(["default", "poll"]),
  pollType: PropTypes.oneOf(["image", "text", "survey"]),
  onClick: PropTypes.func,
  variant: PropTypes.string,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  id: PropTypes.number,
  className: PropTypes.string,
};
