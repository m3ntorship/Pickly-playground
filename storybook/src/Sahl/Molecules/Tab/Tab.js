import React from "react";
import PropTypes from "prop-types";
import Radio from "../../Atoms/Radio/Radio";
import ImageIcon from "../../assets/icons/Image-outlined";
import FileIcon from "../../assets/icons/File-outlined";
import PollIcon from "../../assets/icons/Poll-outlined";

export default function Tab(props) {
  const {
    disabled = false,
    label,
    type = "default",
    onClick,
    pollType,
    className,
    selected,
    value,
  } = props;

  // Style Classes
  let buttonClasses = [
    "h-19",
    "flex",
    "items-center",
    "rounded-full",
    "py-1.5",
    "px-7",
    "focus:outline-none",
    "hover:bg-grey-shd6",
  ];
  disabled ? buttonClasses.push("disabled:opacity-30") : buttonClasses.push("");

  // Handlers

  if (selected) {
    console.log(selected, "[Tab.js]");
    buttonClasses = [
      ...buttonClasses,
      "bg-white",
      "border",
      "border-solid",
      "border-grey-dark",
    ];
  } else {
    buttonClasses = [...buttonClasses, "bg-grey-bg"];
  }

  // Markup
  const pollElements = (
    <>
      <Radio
        value={value}
        name={`${type} tab`}
        checked={selected}
        onChange={onClick}
        className="mr-4"
      />
      {/* ////// Image /////// */}
      {pollType === "image" && (
        <ImageIcon width="w-4" height="h-4" className="mr-2" />
      )}
      {pollType === "text" && (
        <PollIcon width="w-4" height="h-4" className="mr-2" />
      )}
      {pollType === "survey" && (
        <FileIcon width="w-4" height="h-4" className="mr-2" />
      )}
    </>
  );

  return (
    <button
      value={value}
      onClick={onClick}
      disabled={disabled}
      className={`${buttonClasses.join(" ")} ${className}`}
    >
      {type === "poll" && pollElements}
      <span className="text-body-small text-dark-grey capitalize">{label}</span>
    </button>
  );
}

Tab.propTypes = {
  type: PropTypes.oneOf(["default", "poll"]),
  pollType: PropTypes.oneOf(["image", "text", "survey"]),
  onClick: PropTypes.func,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  className: PropTypes.string,
};
