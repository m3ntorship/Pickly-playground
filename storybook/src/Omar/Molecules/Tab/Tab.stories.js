import React from "react";
import { Tab, SIZE, ICONS } from "./Tab";

export default {
  title: "Omar/Molecules/Tab",
  component: Tab,
  argsTypes: {
    ButtonSize: {
      control: {
        type: "select",
        options: [SIZE.BUTTONSIZE.DEFAULT, SIZE.BUTTONSIZE.SMALL],
      },
    },
    Icons: {
      control: {
        type: "select",
        options: [ICONS.IMAGE, ICONS.TEXT, ICONS.DOCUMENT],
      },
    },
  },
};

const Template = (args) => <Tab {...args} />;
export const Default = Template.bind({});
Default.args = {
  ButtonSize: SIZE.BUTTONSIZE.DEFAULT,
  Icons: ICONS.IMAGE,
  LabelText: "Label",
  disabled: false,
};
