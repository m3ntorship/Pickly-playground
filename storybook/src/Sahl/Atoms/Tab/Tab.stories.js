import React from "react";

import Tab from "./Tab";

const defaultExport = {
  title: "Sahl/Atoms/Tab",
  component: Tab,
};

export default defaultExport;

const Template = (args) => <Tab {...args} />;

export const Default = Template.bind({});
Default.args = {
  type: "default",
  label: 'Label',
  disabled: 'false',
};

/* export const PollImage = Template.bind({});
PollImage.args = {
  type: "poll",
  pollType: "image",
  label: 'Label'
};

export const PollText = Template.bind({});
PollText.args = {
  type: "poll",
  pollType: "text",
  label: 'Label'
};

export const PollSurvey = Template.bind({});
PollSurvey.args = {
  type: "poll",
  pollType: "survey",
  label: 'Label'
};
 */