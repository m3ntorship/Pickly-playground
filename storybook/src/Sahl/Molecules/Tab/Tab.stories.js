import React from "react";

import Tab from "./Tab";

const defaultExport = {
  title: "Sahl/Molecules/Tab",
  component: Tab,
};

export default defaultExport;

const Template = (args) => <Tab {...args} />;

export const Default = Template.bind({});
Default.args = {
  type: "default",
  label: "Label",
  disabled: false,
};
