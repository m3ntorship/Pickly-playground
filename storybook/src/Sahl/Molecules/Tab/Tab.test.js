import React from "react";
import renderer from "react-test-renderer";
import Tab from "./Tab";

describe("Tab Component", () => {
  it("should render with label when passing label prop ", () => {
    const tree = renderer.create(<Tab label="label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with only label when passing type='default'", () => {
    const tree = renderer.create(<Tab label="label" type="default" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with radio button when passing type='poll' ", () => {
    const tree = renderer.create(<Tab label="label" type="poll" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render disabled when passing disabled prop ", () => {
    const tree = renderer.create(<Tab label="label" disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with radio button & image icon when passing type='poll' & pollType='image'", () => {
    const tree = renderer
      .create(<Tab label="label" type="poll" pollType="image" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with radio button & text icon when passing type='poll' & pollType='text'", () => {
    const tree = renderer
      .create(<Tab label="label" type="poll" pollType="text" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with radio button & file icon when passing type='poll' & pollType='survey'", () => {
    const tree = renderer
      .create(<Tab label="label" type="poll" pollType="survey" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
