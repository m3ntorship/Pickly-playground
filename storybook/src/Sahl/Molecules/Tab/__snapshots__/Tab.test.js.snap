// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Component should render disabled when passing disabled prop  1`] = `
<button
  className="h-19 flex items-center rounded-full py-1.5 px-7 focus:outline-none hover:bg-grey-shd6 disabled:opacity-30 bg-grey-bg undefined"
  disabled={true}
>
  <span
    className="text-body-small text-dark-grey capitalize"
  >
    label
  </span>
</button>
`;

exports[`Tab Component should render with label when passing label prop  1`] = `
<button
  className="h-19 flex items-center rounded-full py-1.5 px-7 focus:outline-none hover:bg-grey-shd6  bg-grey-bg undefined"
  disabled={false}
>
  <span
    className="text-body-small text-dark-grey capitalize"
  >
    label
  </span>
</button>
`;

exports[`Tab Component should render with only label when passing type='default' 1`] = `
<button
  className="h-19 flex items-center rounded-full py-1.5 px-7 focus:outline-none hover:bg-grey-shd6  bg-grey-bg undefined"
  disabled={false}
>
  <span
    className="text-body-small text-dark-grey capitalize"
  >
    label
  </span>
</button>
`;

exports[`Tab Component should render with radio button & file icon when passing type='poll' & pollType='survey' 1`] = `
<button
  className="h-19 flex items-center rounded-full py-1.5 px-7 focus:outline-none hover:bg-grey-shd6  bg-grey-bg undefined"
  disabled={false}
>
  <label
    className="flex items-center justify-end relative cursor-pointer rounded-full group mr-4"
  >
    <input
      checked="false"
      className="opacity-0 relative z-10"
      disabled={false}
      name="poll tab"
      type="radio"
    />
    <div
      className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
    >
      <div
        className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
      />
    </div>
    <span
      className="block"
    />
  </label>
  <svg
    className="w-4 h-4 mr-2"
    fill="#5B6366"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19.903 8.586C19.854 8.48 19.793 8.379 19.707 8.293L13.707 2.293C13.621 2.207 13.52 2.146 13.414 2.097C13.384 2.083 13.352 2.075 13.32 2.064C13.236 2.036 13.15 2.018 13.061 2.013C13.04 2.011 13.021 2 13 2H6C4.897 2 4 2.897 4 4V20C4 21.103 4.897 22 6 22H18C19.103 22 20 21.103 20 20V9C20 8.979 19.989 8.96 19.987 8.938C19.982 8.849 19.965 8.763 19.936 8.679C19.926 8.647 19.917 8.616 19.903 8.586ZM16.586 8H14V5.414L16.586 8ZM6 20V4H12V9C12 9.553 12.447 10 13 10H18L18.002 20H6Z"
      fill="#5B6366"
    />
    <path
      d="M8 12H16V14H8V12ZM8 16H16V18H8V16ZM8 8H10V10H8V8Z"
      fill="#5B6366"
    />
  </svg>
  <span
    className="text-body-small text-dark-grey capitalize"
  >
    label
  </span>
</button>
`;

exports[`Tab Component should render with radio button & image icon when passing type='poll' & pollType='image' 1`] = `
<button
  className="h-19 flex items-center rounded-full py-1.5 px-7 focus:outline-none hover:bg-grey-shd6  bg-grey-bg undefined"
  disabled={false}
>
  <label
    className="flex items-center justify-end relative cursor-pointer rounded-full group mr-4"
  >
    <input
      checked="false"
      className="opacity-0 relative z-10"
      disabled={false}
      name="poll tab"
      type="radio"
    />
    <div
      className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
    >
      <div
        className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
      />
    </div>
    <span
      className="block"
    />
  </label>
  <svg
    className="w-4 h-4 mr-2"
    fill="#5B6366"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7.49902 11C8.32745 11 8.99902 10.3284 8.99902 9.5C8.99902 8.67157 8.32745 8 7.49902 8C6.6706 8 5.99902 8.67157 5.99902 9.5C5.99902 10.3284 6.6706 11 7.49902 11Z"
      fill="#5B6366"
    />
    <path
      d="M10.499 14L8.99902 12L5.99902 16H8.99902H11.999H17.999L13.499 10L10.499 14Z"
      fill="#5B6366"
    />
    <path
      d="M19.999 4H3.99902C2.89602 4 1.99902 4.897 1.99902 6V18C1.99902 19.103 2.89602 20 3.99902 20H19.999C21.102 20 21.999 19.103 21.999 18V6C21.999 4.897 21.102 4 19.999 4ZM3.99902 18V6H19.999L20.001 18H3.99902Z"
      fill="#5B6366"
    />
  </svg>
  <span
    className="text-body-small text-dark-grey capitalize"
  >
    label
  </span>
</button>
`;

exports[`Tab Component should render with radio button & text icon when passing type='poll' & pollType='text' 1`] = `
<button
  className="h-19 flex items-center rounded-full py-1.5 px-7 focus:outline-none hover:bg-grey-shd6  bg-grey-bg undefined"
  disabled={false}
>
  <label
    className="flex items-center justify-end relative cursor-pointer rounded-full group mr-4"
  >
    <input
      checked="false"
      className="opacity-0 relative z-10"
      disabled={false}
      name="poll tab"
      type="radio"
    />
    <div
      className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
    >
      <div
        className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
      />
    </div>
    <span
      className="block"
    />
  </label>
  <svg
    className="w-4 h-4 mr-2"
    fill="none"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 11H14V13H7V11ZM7 7H17.97V9H7V7ZM7 15H20V17H7V15ZM4 4H6V20H4V4Z"
      fill="#5B6366"
    />
  </svg>
  <span
    className="text-body-small text-dark-grey capitalize"
  >
    label
  </span>
</button>
`;

exports[`Tab Component should render with radio button when passing type='poll'  1`] = `
<button
  className="h-19 flex items-center rounded-full py-1.5 px-7 focus:outline-none hover:bg-grey-shd6  bg-grey-bg undefined"
  disabled={false}
>
  <label
    className="flex items-center justify-end relative cursor-pointer rounded-full group mr-4"
  >
    <input
      checked="false"
      className="opacity-0 relative z-10"
      disabled={false}
      name="poll tab"
      type="radio"
    />
    <div
      className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
    >
      <div
        className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
      />
    </div>
    <span
      className="block"
    />
  </label>
  <span
    className="text-body-small text-dark-grey capitalize"
  >
    label
  </span>
</button>
`;
